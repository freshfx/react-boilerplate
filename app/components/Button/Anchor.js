import styled from '@emotion/styled'

import buttonStyles from './buttonStyles'

const Anchor = styled.a`
  ${buttonStyles};
`

export default Anchor
