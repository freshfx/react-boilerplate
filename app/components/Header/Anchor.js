import styled from '@emotion/styled'
import NormalA from 'components/Anchor'

const Anchor = styled(NormalA)`
  padding: 2em 0;
`

export default Anchor
