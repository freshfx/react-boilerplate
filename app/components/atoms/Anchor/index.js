/**
 * A link to a certain page, an anchor tag
 */

import styled from '@emotion/styled'

const Anchor = styled.a`
  color: #41addd;

  &:hover {
    color: #6cc0e5;
  }
`

export default Anchor
