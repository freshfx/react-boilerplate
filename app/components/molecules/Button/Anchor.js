import styled from '@emotion/styled'

import buttonStyles from 'components/atoms/Button/buttonStyles'

const Anchor = styled.a`
  ${buttonStyles};
`

export default Anchor
