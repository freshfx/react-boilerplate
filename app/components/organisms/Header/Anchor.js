import styled from '@emotion/styled'

import NormalA from 'components/atoms/Anchor'

const Anchor = styled(NormalA)`
  padding: 2em 0;
`

export default Anchor
