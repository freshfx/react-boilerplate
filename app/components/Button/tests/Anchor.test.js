import React from 'react'
import {shallow} from 'enzyme'

import Anchor from '../Anchor'

describe('<A />', () => {
  it('should render an <a> tag', () => {
    const renderedComponent = shallow(<Anchor />)
    expect(renderedComponent.type()).toEqual('a')
  })

  it('should have a className attribute', () => {
    const renderedComponent = shallow(<Anchor />)
    expect(renderedComponent.prop('className')).toBeDefined()
  })

  it('should adopt a valid attribute', () => {
    const id = 'test'
    const renderedComponent = shallow(<Anchor id={id} />)
    expect(renderedComponent.prop('id')).toEqual(id)
  })

  it('should not adopt an invalid attribute', () => {
    const renderedComponent = shallow(<Anchor attribute="test" />)
    expect(renderedComponent.prop('attribute')).toBeUndefined()
  })
})
