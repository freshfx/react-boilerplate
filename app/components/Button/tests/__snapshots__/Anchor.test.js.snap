// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Anchor should match snapshot 1`] = `
.emotion-0 {
  display: inline-block;
  box-sizing: border-box;
  padding: 0.25em 2em;
  -webkit-text-decoration: none;
  text-decoration: none;
  border-radius: 4px;
  -webkit-font-smoothing: antialiased;
  -webkit-touch-callout: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  outline: 0;
  font-family: 'Helvetica Neue',Helvetica,Arial,sans-serif;
  font-weight: bold;
  font-size: 16px;
  border: 2px solid #41addd;
  color: #41addd;
}

.emotion-0:active {
  background: #41addd;
  color: #fff;
}

<a
  class="emotion-0 emotion-1"
  href="http://mxstbr.com/"
/>
`;
